code/game/gamemodes/cult/runes.dm

```diff
@@ -340,144 +340,101 @@ var/list/teleport_runes = list()
-			if(M.mind in sacrificed)
-				continue
-		if(!iscultist(M))
-			possible_targets.Add(M)
-	var/mob/offering
-	if(possible_targets.len > 1) //If there's more than one target, allow choice
-		offering = input(user, "Choose an offering to sacrifice.", "Unholy Tribute") as null|anything in possible_targets
-		if(!Adjacent(user) || !src || QDELETED(src) || user.incapacitated())
-			return
-	else if(possible_targets.len) //Otherwise, if there's a target at all, pick the only one
-		offering = possible_targets[possible_targets.len]
-	if(!offering)
-		rune_in_use = 0
-		return
 	if(((ishuman(offering) || isrobot(offering)) && offering.stat != DEAD) || is_sacrifice_target(offering.mind)) //Requires three people to sacrifice living targets
 		if(invokers.len < 3)
 			for(var/M in invokers)
 				to_chat(M, "<span class='cultitalic'>[offering] is too greatly linked to the world! You need three acolytes!</span>")
 			fail_invoke()
 			log_game("Sacrifice rune failed - not enough acolytes and target is living")
-			rune_in_use = 0
 			return
-	visible_message("<span class='warning'>[src] pulses blood red!</span>")
-	color = rgb(126, 23, 23)
-	..()
-	sac(invokers, offering)
-	color = initial(color)
 
-/obj/effect/rune/sacrifice/proc/sac(var/list/invokers, mob/living/T)
 	var/sacrifice_fulfilled
 	var/datum/game_mode/cult/cult_mode = SSticker.mode
-	if(T)
-		if(isdog(T)) // Doggos are the best, but not cats
-			for(var/M in invokers)
-				var/mob/living/L = M
-				to_chat(L, "<span class='cultlarge'>\"Even I have standards, such as they are!\"</span>")
-				if(L.reagents)
-					L.reagents.add_reagent("hell_water", 2)
-		if(T.mind)
-			sacrificed.Add(T.mind)
-			if(is_sacrifice_target(T.mind))
-				sacrifice_fulfilled = 1
-		new /obj/effect/temp_visual/cult/sac(loc)
-		if(SSticker && SSticker.mode && SSticker.mode.name == "cult")
-
-			cult_mode.harvested++
-		for(var/M in invokers)
-			if(sacrifice_fulfilled)
-				to_chat(M, "<span class='cultlarge'>\"Yes! This is the one I desire! You have done well.\"</span>")
-				cult_mode.additional_phase()
-			else
-				if(ishuman(T) || isrobot(T))
-					to_chat(M, "<span class='cultlarge'>\"I accept this sacrifice.\"</span>")
-				else
-					to_chat(M, "<span class='cultlarge'>\"I accept this meager sacrifice.\"</span>")
-		if(T.mind)
-			var/obj/item/soulstone/stone = new /obj/item/soulstone(get_turf(src))
-			stone.invisibility = INVISIBILITY_MAXIMUM //so it's not picked up during transfer_soul()
-			if(!stone.transfer_soul("VICTIM", T, usr)) //If it cannot be added
-				qdel(stone)
-			if(stone)
-				stone.invisibility = 0
-			if(!T)
-				rune_in_use = 0
-				return
-		if(isrobot(T))
-			playsound(T, 'sound/effects/EMPulse.ogg', 100, 1)
-			T.dust() //To prevent the MMI from remaining
+	if(offering.mind)
+		sacrificed.Add(offering.mind)
+		if(is_sacrifice_target(offering.mind))
+			sacrifice_fulfilled = 1
+	new /obj/effect/temp_visual/cult/sac(loc)
+	if(SSticker && SSticker.mode && SSticker.mode.name == "cult")
```

Comment:
SSTicker should be checked earlier. Since you already take the mode.
Or better yet move `var/datum/game_mode/cult/cult_mode = SSticker.mode` to after the if or refactor it a bit.

Reply:
Well that's where SSticker is checked in the current sacrifice rune, and the current sacrifice rune properly works, and my testing showed it still did with my refactor. I'm completely inexperienced with SSticker and modes, if you know exactly where it should go please share, otherwise I'm not gonna toy with it.

Reply:
Currently at the top it makes a variable cult_mode which takes the tickers game mode and casts it to an type.
Here in this if you check if SSTicker is not null and SSTicker.mode is not null and if SSTicker.mode is a cult objective. The first check, if Ticker is null is caught as an exception if it is null due to you using SSTicker above. The other 2 are tested because you parse the mode to the cult mode. If the mode is not cult then the parsed var (cult_mode) will fail if you use it. So you can do istype(cult_mode).

Reply:
After chatting about it on the Discord, I get a bit more of what you're saying, but I still don't think I've got the chops for handling this. As it is now, the code works, and it's the code we already have, unchanged. I'll leave this refactor for someone that really knows what they're doing. This PR was about combing two existing runes into one, while fixing a couple functionality bugs.